src/js/actions/transform.js

```diff
@@ -231,7 +235,42 @@ define(function (require, exports) {
         // Used to calculate the new top/left positions of all layers in relation to top/left
         // of the group
         var overallWidthRatio = size.w ? size.w / overallBounds.width : 0,
-            overallHeightRatio = size.h ? size.h / overallBounds.height : 0;
+            overallHeightRatio = size.h ? size.h / overallBounds.height : 0,
+            overallLeftOffset = 0,
+            overallTopOffset = 0,
+            direction;
+
+        // This is the case if the width input was changed
```

Comment:
Here's the most important request: please write a paragraph-length explanation of how this works using a `// comment`. The algorithm for this was complicated enough that we all screwed it up at least a few times, but you've now cracked the code. Explain this to your future self! (And also the reviewers.)